### PR TITLE
Fix upgrades from the old website!

### LINKED ISSUE #485: Only old site shows until browser data cleared
Hey,

I think I probably last visited preactjs.com several months ago. Certainly before Preact X documentation was present.

It seems I had a service worker installed that happily presented me the old site repeatedly, until I manually cleared site data stored in my browser and the new website finally appeared. Very nice it is, too!

A screenshot of the old site, presented despite its age:
Screenshot from 2019-11-24 09-57-35

Post-cookie clear, cache clear, and SW unregister (Chrome 'Clear site data' button) - and after I loaded the new site in a new tab - here's the old tab, on the Service Worker page in Chrome DevTools:

Screenshot from 2019-11-24 10-01-52

Note the 'clients' list including both the old (versionless) getting started guide and the new v10 what's new page.

When is the old service worker invalidated? I spent a good 25 mins futzing around in the browser, trying to understand why the live site had nothing about v10 compared to this repository. I was persistently presented the old site.

### PR DESCRIPTION
**Pull request by developit**
This fixes #485 

/cc @prateekbh

### CODE REVIEW

File: src/sw.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import 'preact-cli/lib/lib/sw.js';
+
+// fix upgrades from the old site
+self.caches.keys().then(keys => {
+	const key = keys.filter(k => /^webpack-offline:/.test(k))[0];
+	if (key) {
+		caches.delete(key);
+		setTimeout(() => {
```
Comment (prateekbh) [inline src/sw.js]: skip waiting in a loop?

File: src/sw.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'preact-cli/lib/lib/sw.js';
+
+// fix upgrades from the old site
+self.caches.keys().then(keys => {
+	const key = keys.filter(k => /^webpack-offline:/.test(k))[0];
+	if (key) {
+		caches.delete(key);
```
Comment (prateekbh) [inline src/sw.js]: .then(() => {skipWaiting....})

**Comment by marvinhagemeister** [comment]
@prateekbh Did you intend to close this PR or was that by mistake?

**Comment by prateekbh** [comment]
woops sorry no, closed it by mistake

File: src/sw.js (hunk 2 above)
Comment (developit) [reply src/sw.js]: @prateekbh - the old SW already did this when a new SW got installed, so it's unnecessary. In local testing, calling skipWaiting() triggers the page to reload itself, which loads the new version.

Comment (developit) [reply src/sw.js]: the old client calls location.reload() when a new SW is installed, which in my testing nicely reloads the page.

**Pull request merged**